README.md

```diff
@@ -547,6 +541,92 @@ In this case:
+
+> ... a user’s keyboard focus point may be kept in the same place as where they clicked, which isn’t intuitive. In layouts where the page changes partially to include a deep-linked modal dialog or other view layer, a user’s focus point could be left in an entirely wrong spot on the page.
+
+App history navigation interception again gives us the tool to fix this.
+
+By default, any navigation that is intercepted and converted into a single-page navigation using `navigateEvent.transitionWhile()` will cause focus to reset to the `<body>` element, or to the first element with the `autofocus=""` attribute set (if there is one). This focus reset will happen in the first animation frame after the promise passed to `transitionWhile()` settles. However, this focus reset will not take place if the user has manually changed focus while the promise was settling, and the element the user changed focus to is still visible and focusable.
+
+This behavior can be customized using the second options argument to `transitionWhile()`:
+
+- `e.transitionWhile(promise, { focusReset: "after-transition" })`: the default behavior, described above.
+- `e.transitionWhile(promise, { focusReset: "manual" })`: does not reset the focus, and leaves it where it is. (Although, it might get [reset anyway](https://html.spec.whatwg.org/#focus-fixup-rule) if the element is removed from the DOM or similar.) The application will manually manage focus changes.
+
+In general, the default behavior is a best-effort attempt at cross-document navigation parity. But if developers invest some extra work, they can do even better:
+
+- Per the above-linked [research by Fable Tech Labs](https://www.gatsbyjs.com/blog/2019-07-11-user-testing-accessible-client-routing/), screen reader users generally prefer focus to be reset to a heading or wrapper element, instead of the `<body>` element. So to get the optimal experience with app history interception, developers should use `autofocus=""` appropriately on such elements.
+
+- For traversals (i.e. cases where `navigateEvent.navigationType === "traverse"`), getting parity with the [back/forward cache experience](https://web.dev/bfcache/) requires restoring focus to the same element that was previously focused when that history entry was active. Unfortunately, this isn't something the browser can do automatically for client-side rendered applications; the notion of "the same element" [is not generally stable](https://github.com/WICG/app-history/issues/190#:~:text=On%20the%20other%20hand%2C%20in%20the%20general%20case%20we%20won%27t%20be%20able%20to%20identify%20%22the%20same%20element%22!) in such cases. So for such cases, using `focusReset: "manual"`, storing some identifier for the currently-focused element in the app history state, and calling `element.focus()` appropriately upon transition could give a better experience.
```

Comment:
Can you comment here when it would be appropriate to call element.focus() after transition? Would you recommend using the transition property, or in a currentchange listener, after the promise resolves, etc.

Reply:
Any of `appHistory.transition.finished.then(updateFocus)`, `appHistory.back().then(updateFocus)` or `appHistory.addEventListener("navigatesuccess", updateFocus)` would work; they have the same timing.

If you wanted to update focus even on failure then you'd do `appHistory.transition.finished.finally(updateFocus)` or `appHistory.back().finally(updateFocus)`. For the event version you'd add handlers for both `navigatesuccess` and `navigateerror`.

You could also integrate it into your specific application logic, e.g.

```js
appHistory.addEventListener("navigate", e => {
  e.transitionWhile((async () => {
    await updateDOMWithSkeleton();
    findElement().focus();
    await updateTheRestOfTheDOM();
  })());
});
```

I'll mention the basics here in the PR.